### ISSUE TITLE
Inserter: Always show patterns

### ISSUE THREAD

**Issue by scruffian** [WordPress/gutenberg#63085]
## What problem does this address?
When a block is selected, and the patterns tab is selected, no patterns are shown:

<img width="1778" alt="Screenshot 2024-07-03 at 12 32 25" src="https://github.com/WordPress/gutenberg/assets/275961/ef58b9e9-e249-455e-963d-24b5e815507e">

This will become more of a problem once https://github.com/WordPress/gutenberg/pull/63059 is merged.

## What is your proposed solution?

We should show all patterns and insert them after the selected block, if the selected block cannot contain a pattern.

**Comment by scruffian** [comment]
Closing in favour of https://github.com/WordPress/gutenberg/issues/61320
